apps/spruce/src/pages/configurePatch/configurePatchCore/useConfigurePatch/index.ts

```diff
@@ -1,169 +1,58 @@
-        selectedBuildVariantTasks: action.variantTasks,
-        selectedAliases: action.aliases,
-      };
-
-    default:
-      throw new Error("Unknown action type");
-  }
-};
-
-const indexToTabMap = [PatchTab.Tasks, PatchTab.Changes, PatchTab.Parameters];
-
-const tabToIndexMap = {
-  [PatchTab.Tasks]: 0,
-  [PatchTab.Changes]: 1,
-  [PatchTab.Parameters]: 2,
-};
+import { ConfigurePatchPageTabs } from "types/patch";
+import { indexToTabMap, tabToIndexMap } from "./constants";
+import { ConfigurePatchState, initialState, reducer } from "./state";
+import { AliasState, VariantTasksState } from "./types";
+import { initializeAliasState, initializeTaskState } from "./utils";
 
 interface HookResult extends ConfigurePatchState {
   setDescription: (description: string) => void;
   setPatchParams: (patchParams: ParameterInput[]) => void;
   setSelectedBuildVariants: (variants: string[]) => void;
   setSelectedBuildVariantTasks: (variantTasks: VariantTasksState) => void;
   setSelectedAliases: (aliases: AliasState) => void;
-  setSelectedTab: React.Dispatch<React.SetStateAction<number>>;
+  setSelectedTab: (tab: ConfigurePatchPageTabs) => void;
 }
 
 const useConfigurePatch = (patch: ConfigurePatchQuery["patch"]): HookResult => {
   const navigate = useNavigate();
-  const location = useLocation();
-  // @ts-expect-error: FIXME. This comment was added by an automated script.
-  const { [slugs.tab]: tab } = useParams<{ [slugs.tab]: PatchTab | null }>();
+  const { [slugs.tab]: urlTab } = useParams<{
+    [slugs.tab]: ConfigurePatchPageTabs;
+  }>();
 
-  const { id, project } = patch;
-  // @ts-expect-error: FIXME. This comment was added by an automated script.
-  const { variants } = project;
+  const { id, project } = patch || {};
+  const { variants } = project || {};
   const [state, dispatch] = useReducer(
     reducer,
     initialState({
-      // @ts-expect-error: FIXME. This comment was added by an automated script.
-      selectedTab: tabToIndexMap[tab || PatchTab.Configure],
+      selectedTab: urlTab || ConfigurePatchPageTabs.Tasks,
     }),
   );
 
-  const { selectedTab } = state;
-
+  // Update the URL when the selected tab changes
   useEffect(() => {
-    const query = parseQueryString(location.search);
     navigate(
       getPatchRoute(id, {
         configure: true,
-        tab: indexToTabMap[selectedTab],
-        ...query,
+        tab: state.selectedTab,
       }),
       { replace: true },
     );
-  }, [selectedTab]); // eslint-disable-line react-hooks/exhaustive-deps
+  }, [state.selectedTab]); // eslint-disable-line react-hooks/exhaustive-deps
 
   useEffect(() => {
     if (patch) {
       dispatch({
         type: "updatePatchData",
         description: patch.description,
+        // @ts-expect-error
         buildVariants: [variants[0]?.name],
         params: patch.parameters,
+        // @ts-expect-error
         variantTasks: initializeTaskState(variants, patch.variantsTasks),
```

Comment:
nit. if you give `variants` a default empty value, maybe you can do:
```suggestion
        buildVariants: variants.length > 0 ? [variants[0].name] : [],
        params: patch.parameters,
        variantTasks: initializeTaskState(variants, patch.variantsTasks),
```